### PR TITLE
Remove no-rfc3779 support

### PR DESCRIPTION
**Pull request by richsalz**
It's always on.

## Checklist
Thank you for your contribution. Please answer the following:

- [x] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)
- [ ] All new public APIs are documented
- [ ] All new public functions have tests
- [ ] All new public functions have fuzzing tests

### CODE REVIEW

**Review by botovq — APPROVED** [review]
I'm fine with doing this, although the RFC 3779 code is of rather low quality even by OpenSSL standards. I documented a few examples in the BUGS section of [our manual][1]. All these functions are still undocumented in OpenSSL/QuicTLS.

[1]: https://man.openbsd.org/X509v3_addr_add_range#BUGS

**Comment by richsalz** [comment]
Ah, I typed into the wrong window.  I was only looking at the Configure code and counting wc|sort |uniq -c ifdefs.
Ugh.  Not sure if still worth merging this tho.

**Comment by botovq** [comment]
It's highly specialized. Since there are pushes towards making use of the RPKI mandatory in the foreseeable future, I'm not sure flipping the default is a good move.

We use a little bit of the API in [rpki-client](https://github.com/rpki-client/rpki-client-portable). There's also [FORT](https://github.com/NICMx/FORT-Validator). There's also a long lingering PR at rust-openssl: https://github.com/sfackler/rust-openssl/pull/2053 not sure what's up with that and if that will ever be merged (I never understood what's merged into rust-openssl and what isn't and why)

Both Fort and rpki-client make use of the parts hooking into the X.509 validation algorithm (which will soon be outdated due to being too strict...). The code is only used if certificates have one the two extensions defined in RFC 3779 and all certificates doing so should have the relevant certificate policy (RFC 6485). But of course any cert can choose to add the extensions and thus exercise this particular pile of garbage. In view of planned updates to the specification, I am going to disable the cert validation part by default in LibreSSL soon. I will also try to upstream that to OpenSSL (since many people use rpki-client with OpenSSL. I will of course also send that your way once the details are settled.

In short, I think whether you keep the knob or not is your choice. It's not too intrusive #ifdefery and it of course makes sense to disable that code if one is only interested in the web PKI.

**Comment by richsalz** [comment]
Thanks for the insight.  Going to close this with no action, but "low quality even by OpenSSL standards" is a phrase I hope to use again :)
